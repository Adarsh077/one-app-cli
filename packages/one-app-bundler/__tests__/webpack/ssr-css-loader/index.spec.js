/*
 * Copyright 2019 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations
 * under the License.
 */

const SSRCSSLoader = require('../../../webpack/loaders/ssr-css-loader');

jest.mock('path', () => ({
  resolve: (dir, filename) => `/path/to/one-app-bundler/webpack/ssr-css-loader/${filename}`,
}));

describe('SSR CSS loader', () => {
  const consoleSpy = jest.spyOn(console, 'log');

  beforeEach(() => consoleSpy.mockClear());

  it('should throw an error when the css-loader is not present', () => {
    expect(() => SSRCSSLoader('no css-loader here!')).toThrowErrorMatchingSnapshot();
    expect(() => SSRCSSLoader('exports = module.exports = require("css-loader/dist/runtime/api.js")();')).toThrowErrorMatchingSnapshot();
    expect(() => SSRCSSLoader('exports = module.exports = require("../../css-loader/dist/runtime/api.js")(true);')).toThrowErrorMatchingSnapshot();
    expect(() => SSRCSSLoader(`
      // some other line
      exports = module.exports = require("../../see-ess-ess-loader/lib/see-ess-ess-base.js")(undefined);
      // imports


      // module
      exports.push([module.id, "/**comment**/.react-select__One___hIKe {\n  position: relative; }\n\n.react-select__also___8ObpT {\n  background-color: #f9f9f9; }\n", ""]);

      // exports
      exports.locals = {
      \t"One": "react-select__One___hIKe",
      \t"also": "react-select__also___8ObpT",
      };
    `)).toThrowErrorMatchingSnapshot();
  });

  it('should remove css-loader\'s css-base, replace its push and export the locals', () => {
    expect(SSRCSSLoader('exports = module.exports = require("../../node_modules/css-loader/dist/runtime/api.js")(undefined);')).toMatchSnapshot();
    expect(SSRCSSLoader('exports = module.exports = require("../../css-loader/dist/runtime/api.js")();')).toMatchSnapshot();
    expect(SSRCSSLoader('exports = module.exports = require("./css-loader/dist/runtime/api.js")();')).toMatchSnapshot();
    expect(SSRCSSLoader('exports = module.exports = require("./css-loader/dist/runtime/api.js")(undefined);')).toMatchSnapshot();
    expect(SSRCSSLoader(`
      // some other line
      exports = module.exports = require("../../css-loader/dist/runtime/api.js")(undefined);
      // imports


      // module
      exports.push([module.id, "/**comment**/.react-select__One___hIKe {\n  position: relative; }\n\n.react-select__also___8ObpT {\n  background-color: #f9f9f9; }\n", ""]);

      // exports
      exports.locals = {
      \t"One": "react-select__One___hIKe",
      \t"also": "react-select__also___8ObpT",
      };
    `)).toMatchSnapshot();
    expect(SSRCSSLoader('exports = module.exports = require("../../node_modules/css-loader/dist/runtime/api.js")(false);')).toMatchSnapshot();
    expect(SSRCSSLoader('exports = module.exports = require("./css-loader/dist/runtime/api.js")(false);')).toMatchSnapshot();
    expect(SSRCSSLoader(`
      // some other line
      exports = module.exports = require("../../css-loader/dist/runtime/api.js")(false);
      // imports


      // module
      exports.push([module.id, "/**comment**/.react-select__One___hIKe {\n  position: relative; }\n\n.react-select__also___8ObpT {\n  background-color: #f9f9f9; }\n", ""]);

      // exports
      exports.locals = {
      \t"One": "react-select__One___hIKe",
      \t"also": "react-select__also___8ObpT",
      };
    `)).toMatchSnapshot();
  });
});
