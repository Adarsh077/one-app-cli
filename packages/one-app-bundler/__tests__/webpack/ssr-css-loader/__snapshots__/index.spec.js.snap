// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 1`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 2`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 3`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 4`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 5`] = `
"
      // some other line
      
      // imports


      // module
      require(\\"/path/to/one-app-bundler/webpack/ssr-css-loader/css-base.js\\")().push([module.id, \\"/**comment**/.react-select__One___hIKe {
  position: relative; }

.react-select__also___8ObpT {
  background-color: #f9f9f9; }
\\", \\"\\"]);

      // exports
      exports = module.exports = {
      	\\"One\\": \\"react-select__One___hIKe\\",
      	\\"also\\": \\"react-select__also___8ObpT\\",
      };
    "
`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 6`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 7`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 8`] = `
"
      // some other line
      
      // imports


      // module
      require(\\"/path/to/one-app-bundler/webpack/ssr-css-loader/css-base.js\\")().push([module.id, \\"/**comment**/.react-select__One___hIKe {
  position: relative; }

.react-select__also___8ObpT {
  background-color: #f9f9f9; }
\\", \\"\\"]);

      // exports
      exports = module.exports = {
      	\\"One\\": \\"react-select__One___hIKe\\",
      	\\"also\\": \\"react-select__also___8ObpT\\",
      };
    "
`;

exports[`SSR CSS loader should throw an error when the css-loader is not present 1`] = `
"could not find the css-loader in
no css-loader here!"
`;

exports[`SSR CSS loader should throw an error when the css-loader is not present 2`] = `
"could not find the css-loader in
exports = module.exports = require(\\"css-loader/dist/runtime/api.js\\")();"
`;

exports[`SSR CSS loader should throw an error when the css-loader is not present 3`] = `
"could not find the css-loader in
exports = module.exports = require(\\"../../css-loader/dist/runtime/api.js\\")(true);"
`;

exports[`SSR CSS loader should throw an error when the css-loader is not present 4`] = `
"could not find the css-loader in

      // some other line
      exports = module.exports = require(\\"../../see-ess-ess-loader/lib/see-ess-ess-base.js\\")(undefined);
      // imports


      // module
      exports.push([module.id, \\"/**comment**/.react-select__One___hIKe {
  position: relative; }

.react-select__also___8ObpT {
  background-color: #f9f9f9; }
\\", \\"\\"]);

      // exports
      exports.locals = {
      	\\"One\\": \\"react-select__One___hIKe\\",
      	\\"also\\": \\"react-select__also___8ObpT\\",
      };
    "
`;
